libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -53,4 +54,11 @@ export class BookSearchComponent implements OnInit {
       this.store.dispatch(clearSearch());
     }
   }
+  searchBooksReason() {
+    if (this.searchData !== '') {
+      this.store.dispatch(searchBooks({ term: this.searchData }));
+    } else {
+      this.store.dispatch(clearSearch());
+    }
+  }
```

Comment:
It would be better to use the reactive forms methods to listen to the value changes of the search term. Please take a look at this documentation:
https://angular.io/guide/reactive-forms
You have the `.valueChanges()` observable that you can use!